src/app/static/src/app/components/ResetConfirmation.vue

```diff
@@ -59,32 +66,48 @@
 
     interface Data {
         waitingForVersion: boolean
+        versionNote: string
     }
 
     export default Vue.extend<Data, unknown, Computed, any>({
         props: ["open", "continueEditing", "cancelEditing"],
         data: function () {
             return {
-                waitingForVersion: false
+                waitingForVersion: false,
+                versionNote: ""
             }
         },
         computed: {
             changesToRelevantSteps: mapGetterByName("stepper", "changesToRelevantSteps"),
             currentVersionId: mapStateProp<ProjectsState, string | null>("projects", state => {
                 return state.currentVersion && state.currentVersion.id;
             }),
+            currentVersionNote: mapStateProp<ProjectsState, string | null>("projects", state => {
+                return state.currentVersion?.note || "";
+            }),
             ...mapGetters(["isGuest"]),
             errorsCount: mapStateProp<ErrorsState, number>("errors", state => {
                 return state.errors ? state.errors.length : 0;
-            })
+            }),
+            handleVersionNote: {
+                get() {
+                    return this.currentVersionNote!
```

Comment:
Is `!` necessary? See comment above.

Produces a compiler warning.

Reply:
> Thanks for this. Some comments inline and a couple of bugs that will need fixes, tests etc:
> 
> * If you create a new version with a comment, and then create another version then the textarea is populated with the previous comment - all good. But if I delete the comment then the old comment still gets saved against the new version.
> * If you add a comment such as `` !"#$%&\'()*+,-./:;<=>?@[\\]^_`{|}~ `` then things seem to go wrong - when you create another version then the dialog is only populated with `!"`. It wouldn't be unexpected for people to use random punctuation in notes, so we'll have to do something about this. Note sure if the note isn't being saved faithfully, or whether it's just rendered incorrectly.

I will investigate this regex like issue.